query_optimizer/ExecutionGenerator.cpp

```diff
@@ -760,16 +760,59 @@ void ExecutionGenerator::convertCreateIndex(
   // CreateIndex is converted to a CreateIndex operator.
   const CatalogRelationInfo *input_relation_info =
       findRelationInfoOutputByPhysical(physical_plan->input());
+  // NOTE: the object pointed to by input_relation is owned by CatalogRelation
   CatalogRelation *input_relation =
       optimizer_context_->catalog_database()->getRelationByIdMutable(
             input_relation_info->relation->getID());
+
+  // check if any index with the specified name already exists
   if (input_relation->hasIndexWithName(physical_plan->index_name())) {
     THROW_SQL_ERROR() << "The relation " << input_relation->getName()
-            << " already has an index named "<< physical_plan->index_name();
-  } else {
-    execution_plan_->addRelationalOperator(
-       new CreateIndexOperator(input_relation, physical_plan->index_name()));
+    << " already has an index named "<< physical_plan->index_name();
+  }
+
+  // assertion: must have at least one attribute specified to build index upon
+  DEBUG_ASSERT(physical_plan->index_attributes().size() > 0);
+
+  // convert attribute references to a vector of pointers to catalog attributes
+  std::vector<const CatalogAttribute*> index_attributes;
+  for (const E::AttributeReferencePtr &attribute : physical_plan->index_attributes()) {
+    const CatalogAttribute *catalog_attribute
+      = input_relation->getAttributeByName(attribute->attribute_name());
+    DEBUG_ASSERT(catalog_attribute != nullptr);
```

Comment:
@cramja Marc, the Resolver::resolveCreateIndex() does this very check and prints the attribute undefined error message. So, I thought I could skip it here.